Sparse checkout: Use `--remote` for `git submodule update`

AVMusorin opened the pull request:
cc @rschu1ze 

Prevents errors like

```
fatal: Fetched in submodule path 'contrib/google-protobuf', but it did not contain 089b89c8d4140f0d49fe4222b047a8ea814bc752. Direct fetching of that commit failed.
````

### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

AVMusorin committed:
Added comment to prevent using --remote to update submodules

robot-ch-test-poll2 commented:
<!-- automatic status comment for PR #56857 from AVMusorin/ClickHouse:fix-update-submodules -->
*This is an automated comment for commit 414718b5134b2c23347b32a45fabe0c798a4123f with description of existing statuses. It's updated for the latest CI running*

[❌ Click here](https://s3.amazonaws.com/clickhouse-test-reports/56857/414718b5134b2c23347b32a45fabe0c798a4123f/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td><a href="https://github.com/ClickHouse/ClickHouse/actions/runs/6891099725">✅ success</a></td></tr>
<tbody>
</table>
</details>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/56857/414718b5134b2c23347b32a45fabe0c798a4123f/ci_running.html">❌ failure</a></td></tr>
<tbody>
</table>

rschu1ze commented:
There is some problem, repro:

```bash
$ git clone https://github.com/ClickHouse/ClickHouse.git
$ cd ClickHouse/
# add '--remote' to contrib/update-submodules.sh
$ ./contrib/update-submodules.sh
# cmake -S . -B build
```

fails with

```
CMake Error at contrib/llvm-project/llvm/CMakeLists.txt:82 (message):
  Requested CMAKE_CXX_STANDARD=14 which is less than the required 17.
```

When one does not add `--remote` and runs the repro again, one will see that the submodule in `contrib/llvm-project/` is on a different commit.

AVMusorin commented:
I went through the documentation and how the update works and yes it is my bad, actually it leads to unexpected behavior. So I added a comment to prevent it in the future.

P.S. also added quotes for '.*path' to support other shells, otherwise, they try to execute this part of the command.
